### PR TITLE
Speedloader Box (M44 x 16)

### PR DESCRIPTION
**Pull request by FungiFellow**
## About the PR

I added Speedloader Boxes for the M44's regular and Marksman Rounds alongside entries for them in the requisition catalog.
## Why / Balance

I've played several rounds using the M44 as my primary weapon, it feels solid but kept running into a certain wall called Ammo Deficiency. 

Currently the only way to reliably use it as a primary is to grab an absurd amount of speedloaders (I held upwards of 26 and still ran into ammo issues). This is a major issue because Requisitions on average only sends maybe 12 or so for a whole operation unless one of them is Goated. 

Marksman rounds are also great but with my current setup I can only ever get a hypothetical max of 13 Loaders (Starting Equipment then every single Loader From Req)

Resupplying M44 rounds is a bit more tedious than other ammo types leading to it being neglected. This will cause an uptick in the amount of M44's used

Resolves https://github.com/RMC-14/RMC-14/issues/3421
## Media
image
image
image
image
Actual Descriptions are:
Speedloader Box (M44 x 16) 
"A box for holding many speedloaders with a carrying handle."  

Speedloader Box (Marksman M44 x 16) 
"A box for holding many speedloaders with a carrying handle, these ones are loaded with Marksman Rounds." 

(I adjusted them after testing for screenshots and am not gonna reboot visual studio again)
- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes

**Changelog**

:cl: 
- add: Added M44 Speedloader Boxes
- add: Cargo Can Order M44 Reg and Marksman Rounds

### CODE REVIEW

File: Resources/Prototypes/_RMC14/Catalog/Fills/Crates/magazine_boxes.yml [hunk 1]
```diff
@@ -162,3 +162,22 @@
...
+
+# M44 Revolver
+- type: entity
+  parent: CMCrateAmmo
+  id: RMCCrateBoxMagazineM44Revolver
+  name: Speedloader box (M44, 16x loaders)
+  components:
+  - type: StorageFill
+    contents:
+    - id: CMBoxMagazineRevolverM44
```
Comment (DrSmugleaf) [inline Resources/Prototypes/_RMC14/Catalog/Fills/Crates/magazine_boxes.yml:174]:
```suggestion
    - id: RMCBoxMagazineRevolverM44
```

File: Resources/Prototypes/_RMC14/Catalog/Fills/Crates/magazine_boxes.yml [hunk 2]
```diff
@@ -162,3 +162,22 @@
...
+
+- type: entity
+  parent: CMCrateAmmo
+  id: RMCCrateBoxMagazineM44RevolverMarksman
+  name: Speedloader box (M44 Marksman, 16x loaders)
+  components:
+  - type: StorageFill
+    contents:
+    - id: CMBoxMagazineRevolverMarksmanM44
```
Comment (DrSmugleaf) [inline Resources/Prototypes/_RMC14/Catalog/Fills/Crates/magazine_boxes.yml:183]:
```suggestion
    - id: RMCBoxMagazineRevolverMarksmanM44
```

File: Resources/Prototypes/_RMC14/Entities/Objects/Weapons/Guns/Ammunition/Boxes/magazine_box_crafting.yml [hunk 3]
```diff
@@ -203,6 +203,36 @@
     state: base_su6
   objectType: Item
 
+# M44
+- type: construction
+  parent: RMC
+  name: Magazine Box (M44 Reg)
+  id: CMBoxMagazineRevolverM44Empty
```
Comment (DrSmugleaf) [inline Resources/Prototypes/_RMC14/Entities/Objects/Weapons/Guns/Ammunition/Boxes/magazine_box_crafting.yml:210]:
```suggestion
  id: RMCBoxMagazineRevolverM44Empty
```

File: Resources/Prototypes/_RMC14/Entities/Objects/Weapons/Guns/Ammunition/Boxes/magazine_box_crafting.yml [hunk 4]
```diff
@@ -203,6 +203,36 @@
...
+  description: A box for holding many speedloaders with a carrying handle.
+  icon:
+    sprite: _RMC14/Objects/Weapons/Guns/Ammunition/Boxes/magazine_boxes.rsi
+    state: base_m54
+  objectType: Item
+
+- type: construction
+  parent: RMC
+  name: Magazine Box (M44 Mark)
+  id: CMBoxMagazineRevolverM44MarksmanEmpty
```
Comment (DrSmugleaf) [inline Resources/Prototypes/_RMC14/Entities/Objects/Weapons/Guns/Ammunition/Boxes/magazine_box_crafting.yml:224]:
```suggestion
  id: RMCBoxMagazineRevolverM44MarksmanEmpty
```

File: Resources/Prototypes/_RMC14/Entities/Objects/Weapons/Guns/Ammunition/Boxes/magazine_box_crafting.yml [hunk 5]
```diff
@@ -203,6 +203,36 @@
...
   objectType: Item
 
+# M44
+- type: construction
+  parent: RMC
+  name: Magazine Box (M44 Reg)
+  id: CMBoxMagazineRevolverM44Empty
+  graph: CMBoxMagazine
+  startNode: start
+  targetNode: CMBoxMagazineRevolverM44Empty
```
Comment (DrSmugleaf) [inline Resources/Prototypes/_RMC14/Entities/Objects/Weapons/Guns/Ammunition/Boxes/magazine_box_crafting.yml:213]:
```suggestion
  graph: RMCBoxMagazine
  startNode: start
  targetNode: RMCBoxMagazineRevolverM44Empty
```

File: Resources/Prototypes/_RMC14/Entities/Objects/Weapons/Guns/Ammunition/Boxes/magazine_box_crafting.yml [hunk 6]
```diff
@@ -203,6 +203,36 @@
...
+    state: base_m54
+  objectType: Item
+
+- type: construction
+  parent: RMC
+  name: Magazine Box (M44 Mark)
+  id: CMBoxMagazineRevolverM44MarksmanEmpty
+  graph: CMBoxMagazine
+  startNode: start
+  targetNode: CMBoxMagazineRevolverM44MarksmanEmpty
```
Comment (DrSmugleaf) [inline Resources/Prototypes/_RMC14/Entities/Objects/Weapons/Guns/Ammunition/Boxes/magazine_box_crafting.yml:227]:
```suggestion
  graph: RMCBoxMagazine
  startNode: start
  targetNode: RMCBoxMagazineRevolverM44MarksmanEmpty
```

File: Resources/Prototypes/_RMC14/Entities/Objects/Weapons/Guns/Ammunition/Boxes/magazine_box_crafting.yml [hunk 7]
```diff
@@ -299,6 +329,18 @@
         amount: 1
         doAfter: 1
 
+    # M44
+    - to: CMBoxMagazineRevolverM44Empty
+      steps:
+      - material: RMCSheetCardboard
+        amount: 1
+        doAfter: 1
+    - to: CMBoxMagazineRevolverM44MarksmanEmpty
```
Comment (DrSmugleaf) [inline Resources/Prototypes/_RMC14/Entities/Objects/Weapons/Guns/Ammunition/Boxes/magazine_box_crafting.yml:338]:
```suggestion
    - to: RMCBoxMagazineRevolverM44Empty
      steps:
      - material: RMCSheetCardboard
        amount: 1
        doAfter: 1
    - to: RMCBoxMagazineRevolverM44MarksmanEmpty
```

File: Resources/Prototypes/_RMC14/Entities/Objects/Weapons/Guns/Ammunition/Boxes/magazine_box_crafting.yml [hunk 8]
```diff
@@ -342,4 +384,10 @@
...
   - node: RMCBoxMagazinePistolSU6Empty
-    entity: RMCBoxMagazinePistolSU6Empty
\ No newline at end of file
+    entity: RMCBoxMagazinePistolSU6Empty
+
+  # M44
+  - node: CMBoxMagazineRevolverM44Empty
+    entity: CMBoxMagazineRevolverM44Empty
+  - node: CMBoxMagazineRevolverM44MarksmanEmpty
+    entity: CMBoxMagazineRevolverM44MarksmanEmpty
```
Comment (DrSmugleaf) [inline Resources/Prototypes/_RMC14/Entities/Objects/Weapons/Guns/Ammunition/Boxes/magazine_box_crafting.yml:393]:
```suggestion
  - node: RMCBoxMagazineRevolverM44Empty
    entity: RMCBoxMagazineRevolverM44Empty
  - node: RMCBoxMagazineRevolverM44MarksmanEmpty
    entity: RMCBoxMagazineRevolverM44MarksmanEmpty
```

File: Resources/Prototypes/_RMC14/Entities/Objects/Weapons/Guns/Ammunition/Boxes/magazine_boxes.yml [hunk 9]
```diff
@@ -632,3 +632,85 @@
...
+  - type: CMItemSlots
+    count: 16
+    slot:
+      whitelist:
+        tags:
+        - RMCSpeedLoaderM44
+
+- type: entity
+  parent: CMBoxMagazineRevolverM44MarksmanEmpty
+  id: CMBoxMagazineRevolverMarksmanM44
```
Comment (DrSmugleaf) [inline Resources/Prototypes/_RMC14/Entities/Objects/Weapons/Guns/Ammunition/Boxes/magazine_boxes.yml:707]:
```suggestion
  parent: RMCBoxMagazineBase
  id: RMCBoxMagazineRevolverM44Empty
  name: speedloader box (M44 x 16)
  description: A box for holding many speedloaders with a carrying handle.
  components:
  - type: Construction
    graph: RMCBoxMagazine
    node: RMCBoxMagazineRevolverM44Empty
  - type: Sprite
    layers:
    - state: base_m44 
    - state: magaz_reg 
      map: [ "enum.CMItemSlotsLayers.Fill" ]
    - state: text_m44 
    - state: base_type_m44_reg 
    - state: base_m44_lid 
      map: [ "lid" ]
    - state: lid_type_m44_reg 
      map: [ "lid_stripe" ]
  - type: CMItemSlots
    count: 16
    slot:
      whitelist:
        tags:
        - RMCSpeedLoaderM44

- type: entity
  parent: RMCBoxMagazineRevolverM44Empty
  id: RMCBoxMagazineRevolverM44
  suffix: Filled
  components:
  - type: CMItemSlots
    startingItem: RMCSpeedLoaderM44
    count: 16
    slot:
      whitelist:
        tags:
        - RMCSpeedLoaderM44

- type: entity
  parent: RMCBoxMagazineBase
  id: RMCBoxMagazineRevolverM44MarksmanEmpty
  name: speedloader box (Marksman M44 x 16)
  description: A box for holding many speedloaders with a carrying handle, these ones are loaded with Marksman Rounds.
  components:
  - type: Construction
    graph: RMCBoxMagazine
    node: RMCBoxMagazineRevolverM44MarksmanEmpty
  - type: Sprite
    layers:
    - state: base_m44 
    - state: magaz_reg 
      map: [ "enum.CMItemSlotsLayers.Fill" ]
    - state: text_m44 
    - state: base_type_m44_mark
    - state: base_m44_lid 
      map: [ "lid" ]
    - state: lid_type_m44_mark
      map: [ "lid_stripe" ]
  - type: CMItemSlots
    count: 16
    slot:
      whitelist:
        tags:
        - RMCSpeedLoaderM44

- type: entity
  parent: RMCBoxMagazineRevolverM44MarksmanEmpty
  id: RMCBoxMagazineRevolverMarksmanM44
```

File: Resources/Prototypes/_RMC14/Catalog/Fills/Crates/magazine_boxes.yml [hunk 10]
```diff
@@ -162,3 +162,22 @@
...
+  name: Speedloader box (M44, 16x loaders)
+  components:
+  - type: StorageFill
+    contents:
+    - id: CMBoxMagazineRevolverM44
+
+- type: entity
+  parent: CMCrateAmmo
+  id: RMCCrateBoxMagazineM44RevolverMarksman
+  name: Speedloader box (M44 Marksman, 16x loaders)
```
Comment (DrSmugleaf) [inline Resources/Prototypes/_RMC14/Catalog/Fills/Crates/magazine_boxes.yml:179]:
```suggestion
  name: Speedloader box (Marksman M44, 16x loaders)
```

**Review by DrSmugleaf — APPROVED** [review]
Thanks for the PR! Some of the ids need to be updated from CM to RMC but otherwise it's good

File: Resources/Prototypes/_RMC14/Entities/Objects/Weapons/Guns/Ammunition/Boxes/magazine_boxes.yml [hunk 11]
```diff
@@ -632,3 +632,85 @@
       whitelist:
         tags:
         - RMCMagazinePistolSU6
+
+# M44 Revolver
+
+- type: entity
+  parent: RMCBoxMagazineBase
```
Comment (DrSmugleaf) [inline Resources/Prototypes/_RMC14/Entities/Objects/Weapons/Guns/Ammunition/Boxes/magazine_boxes.yml:639]:
```suggestion
  parent: CMBoxMagazineBase
```

**Review by DrSmugleaf — CHANGES_REQUESTED** [review]

**Comment by FungiFellow** [comment]
> Thanks for the PR! Some of the ids need to be updated from CM to RMC but otherwise it's good

I did think it was a bit odd that the assets had that, will update similar things on future PRs

**Comment by FungiFellow** [comment]
I found the thing that keeps breaking this `id: CMBoxMagazineBase` I gotta adjust all the magazine boxes to RMC

**Comment by DrSmugleaf** [comment]
> I found the thing that keeps breaking this `id: CMBoxMagazineBase` I gotta adjust all the magazine boxes to RMC

You don't need to in this PR but feel free to make a separate one for it once it's merged

**Review by DrSmugleaf — APPROVED** [review]

**Pull request merged**